applications/StructuralMechanicsApplication/python_scripts/element_sensitivity_domain_integration_process.py

```diff
@@ -0,0 +1,109 @@
+# Importing the Kratos Library
+import KratosMultiphysics
+
+def Factory(settings, model):
+    if not isinstance(settings, KratosMultiphysics.Parameters):
+        raise Exception("Expected input shall be a Parameters object, encapsulating a json string")
+    return ElementSensitivityDomainIntegrationProcess(model, settings["Parameters"])
+
+class ElementSensitivityDomainIntegrationProcess(KratosMultiphysics.Process):
+    """
+        This class integrates scalar element sensitivities (material and cross-section
+        properties like CROSS_AREA or YOUNGS_MODULUS) within defined domains.
+        The integration domains are defined by sub model parts of the sensitivity model part.
+    """
+
+    def __init__(self, model, params):
+        """ The default constructor of the class
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        model -- the model contaning the model_parts
+        settings -- Kratos parameters containing solver settings.
+        """
+        KratosMultiphysics.Process.__init__(self)
+
+        ## Settings string in json format
+        default_parameters = KratosMultiphysics.Parameters("""{
+            "help"                             : "This class integrates element sensitivities within domains defined by sub model parts",
+            "element_sensitivity_variables"    : [],
+            "model_part_name"                  : "",
+            "sensitivity_model_part_name"      : "",
+            "sensitivity_sub_model_part_list"  : []
+        }""")
+
+        ## Overwrite the default settings with user-provided parameters
+        params.ValidateAndAssignDefaults(default_parameters)
+        self.params = params
+        self.model  = model
+
+    def ExecuteInitialize(self):
+        """ This method is executed at the begining to initialize the process
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+
+        model_part_name = self.params["model_part_name"].GetString()
+        sensitivity_model_part_name = self.params["sensitivity_model_part_name"].GetString()
+        sensitivity_sub_model_part_names = [ self.params["sensitivity_sub_model_part_list"][i].GetString() for i in range( 0,self.params["sensitivity_sub_model_part_list"].size() ) ]
+
+        # Get sensitivity model part
+        if (sensitivity_model_part_name != ""):
+            self.sensitivity_model_part = self.model[model_part_name].GetSubModelPart(sensitivity_model_part_name)
+        else:
+            self.sensitivity_model_part = self.model[model_part_name]
+
+        # Get defined sub model parts of sensitivity model part as integration domains
+        self.sensitivity_sub_model_parts = []
+        if len(sensitivity_sub_model_part_names) is 0:
+            self.sensitivity_sub_model_parts.append(self.sensitivity_model_part)
+        else:
+            for mp_name in sensitivity_sub_model_part_names:
```

Comment:
why don't you do all the model-part-saving in the constructor?
Then you don't have to save the model

(Note: It is ensure in the AnalysisStage that the ModelParts will be read by the time the processes are constructed)
Not sure though if you do sth different for the sensitivity-model-parts

Reply:
I will move the complete code within `ExecuteInitialize` towards the constructor.